### DISCUSSION TITLE
Feature: Mascot and Achievement Rewards

### DISCUSSION THREAD

**Discussion by rabbids4eva** [RetroAchievements/RAWeb#735] [Ideas]
There should be some unique rewards for completing all the game's cheevos or completing a specific achievement, in a similar way Xbox 360 games give you Avatar items in some titles. 

Oh, and if you know Nintendo's Mii or Xbox avatars, RA could have something similar. Mascots, with unlockable clothes, furniture etc.
You could unlock them by earning enough points from cheevos, participating in weekly achievement challenges and leaderboards, making achievements and through earning achievements in supported games, including special rewards for mastering games on Hardcore. The rewards can be either based on a specific game or in general.

**Comment by rabbids4eva** [comment]
Example: Earn All Achievements in Columns (Genesis) = A Cool Columns themed wallpaper for the room
Example: Earn All Achievements in Destruction Derby 64 (N64) = Get a special interactive prop for your Mascot/Avatar to mess around with.

**Comment by rabbids4eva** [comment]
New examples
Unlock "Kiya from the past" in MediEvil II (PS1) = The golden armor for your mascot/avatar
Master Pac-Man (2600) = A Pac-Man 2600 t-shirt
Unlock "Better Know Them Sevens" in Number Munchers (Apple II) = A "Better Know Them Sevens" custom pyjamas
Unlock "The Headless One" in Paperboy (NES) = A new custom interactivity: headless mode. Makes your mascot / avatar's head detachable in the room. Fully interactive.
Master ~Unlicensed~ Hong Kong '97 (SNES) = A "deng" T-shirt
Unlock "Shopping Friend" in Animal Forest (N64) = A giant Tom Nook plushie
Master MediEvil (PS1) = The statue of Sir Daniel Fortesque

**Comment by ScottFromDerby** [comment]
What sort of reward types are there?
* Wallpaper (Profile)
* Badge
* Others?

What's a mascot? How do we define it? Is it 3d or just an image? What's furniture? How do you imagine it looking?

Is every game allowed one? Would we have quality control implications?

**Comment by FluffyEmmy** [comment]
This is actually a pretty cool idea. Call it like a RAvatar or something. I would imagine it like a basic base sprite, or pixel art, of a man or woman. A specific achievement or mastery (hardcore only) can unlock a shirt, pants, and headgear to overlay the sprite. A decorative item next to (or around) the the sprite, and background behind the sprite. It would be optional and be displayed as either your profile picture, or an image on the sidebar.

As for unlocking them, it should be a limit of one per game and the dev can choose which type of item and if it's a specific achievement (and which one) or mastery. Also for special events, like top AotW winners or something. Maybe even a special item for users with Developer or higher status.

I could see my sets already:
- Pokemon Pinball (Complete Pokedex): A background as if the avatar is the Pokemon in the display area.
- \~Bonus\~ Pokemon RB (Joey Challenge Champion): A hat based on the GCS youngster sprite.
- Halo 2600 (Covenant Defeated): A Master Chief helmet based on the sprite.
- Number Munchers (Master): A decoration with the muncher's open mouth eating the avatar (almost like a frame)
- Tomba (Fashionista): Tomba's basic green pants.

**Comment by rabbids4eva** [comment]
ViperZang explains it well, ScottFromDerby. And ViperZang, we can have either from Mastery or special rewards from specific achievements or tasks in the game. And can be more than one item.

**Comment by Zenjir0** [comment]
I like the idea of the RAvatars being 16-bit or just having a pixel style motif. Like old style Gaia-Online or better yet like the Scott Pilgrim game, art done by Paul Robertson.

**Comment by ScottFromDerby** [comment]
I want this so bad, it's going to take quite a bit of design and consideration to get it going

![image](https://media1.giphy.com/media/X8ZWqLgJo8oFAUhScj/giphy.gif?cid=ecf05e479n28a8obf2op1c8caj8vns8mp3ossfweghf2m2hb&rid=giphy.gif)

![image](https://i.pinimg.com/originals/90/dd/16/90dd16a3041c5aa28ec114106000aefc.gif)

I imagine them being 16-bit constrained but with extras that can make-up an image

1. Are they static avatar images or animated?
2. How do we store "unlocked" assets? (Format, name, texture)
3. How do we support adding new assets? (Upload page, rules to unlock (always achievement bound?)
4. Can we get a similar effect by layering images?
5. Should we layer these images offline and draw a final result, or draw all images on top of eachother with transparency? (Can we do that with html/css?)
